### PR TITLE
Reorganise the "What to look out for" section

### PR DESCRIPTION
**Pull request by josebolos**
This divides the suggestions into 6 big areas, in alphabetical order: Accessibility, Complexity, Performance, Scope, Security, and Syntax.

1333296ee2e03b492e45ae3bdbca7bdd

Also includes:
* Make the text more consistent.
* Clarify the meaning of some terms.
* Replace ambiguous terms (unaccessible code -> accessibility problems).
* Add mentions to security and security tools, potential vulnerabilities, and unnecessary dependencies.

### CODE REVIEW

File: practices/code-review.md [hunk 1]
```diff
@@ -60,11 +60,30 @@ Here's the general process that we use:
...
 
-* Are there syntactic inconsistencies the linter did not catch?
-* Overly complex code.
-* PRs which are non-atomic and should be split into separate PRs, e.g. a new UI feature combined with an unrelated configuration change.
-* Code which could be split up into reusable modules.
-* Non-performant code. Suggest testing with [WebPageTest](https://www.webpagetest.org/) and/or other tools.
-* Inaccessible code. Suggest testing with [pa11y](https://github.com/pa11y) and/or other tools.
+* **Accessibility**
+
+  * Accessibility problems. Suggest testing with [pa11y](https://github.com/pa11y) and/or other tools.
```
Comment (hollsk) [inline practices/code-review.md]: You could link to the accessibility checklist here (or the house style)

File: practices/code-review.md [hunk 2]
```diff
@@ -60,11 +60,30 @@ Here's the general process that we use:
...
     * [internal-only pa11y dashboard](http://pa11y.springernature.com/)
-* Do you see something which may have wider ramifications than the author anticipated? Double check they are aware which can of worms they have just opened :)
+
+* **Complexity**
+
+  * Overly complex code: code that solves problems we don't have, or makes things more complex instead of simplifying things.
```
Comment (hollsk) [inline practices/code-review.md]: "... makes things more complex instead of less."

(removes vague 'things', shortens sentence 😉)

Comment (hollsk) [reply practices/code-review.md]: I know all this stuff was in the original, but hey ho ☺️

File: practices/code-review.md (hunk 1 above)
Comment (josebolos) [reply practices/code-review.md]: I forgot the links! 🤦

File: practices/code-review.md (hunk 2 above)
Comment (josebolos) [reply practices/code-review.md]: Actually I added that bit, so good point :+1:

**Pull request merged**